client/state/sites/selectors.js

```diff
@@ -517,3 +518,95 @@ export function isCurrentSitePlan( state, siteId, planProductId ) {
 
 	return sitePlan.product_id === planProductId;
 }
+
+export function siteCanUpdateFiles( state, siteId ) {
+	if ( ! siteId ) {
+		return null;
+	}
+
+	if ( ! isJetpackSite( state, siteId ) ) {
+		return null;
+	}
+
+	if ( ! siteHasMinimumJetpackVersion( state, siteId ) ) {
+		return false;
+	}
+
+	const isMultiNetwork = getSiteOption( state, siteId, 'is_multi_network' );
+
+	if ( isMultiNetwork ) {
+		return false;
+	}
+
+	if ( ! isMainNetworkSite( state, siteId ) ) {
+		return false;
+	}
+
+	const fileModDisabled = getSiteOption( state, siteId, 'file_mod_disabled' );
+
+	if (
+		fileModDisabled &&
+		(
+			-1 < fileModDisabled.indexOf( 'disallow_file_mods' ) ||
+			-1 < fileModDisabled.indexOf( 'has_no_file_system_write_access' )
+		)
+	) {
+		return false;
+	}
+
+	return true;
+}
+
+export function isMainNetworkSite( state, siteId ) {
+	if ( ! siteId ) {
+		return null;
+	}
+
+	const site = getSite( state, siteId );
+	const isMultiNetwork = getSiteOption( state, siteId, 'is_multi_network' );
+
+	if ( isMultiNetwork ) {
+		return false;
+	}
+
+	if ( site.is_multisite === false ) {
+		return true;
+	}
+
+	if ( site.is_multisite ) {
```

Comment:
Maybe an early return here to avoid the indented block?

```js
if ( ! site.is_multisite ) {
	return false;
}
```

Kinda confusing that the false comparison above returns `true`, but if it makes it here and it's falsey, the selector returns `false`. 😕